### PR TITLE
drivers: sbs_gauge: Add support for AtRate properties

### PR DESCRIPTION
**Pull request by madhurimaparuchuri**
BatteryMode(w), AtRate(r/w), AtRateTimeToFull(r), AtRateTimeToEmpty(r) and AtRateOK(r)

### CODE REVIEW

File: include/zephyr/drivers/fuel_gauge.h [hunk 1]
```diff
@@ -37,45 +37,53 @@ extern "C" {
 #define FUEL_GAUGE_AVG_CURRENT 0
 
 /** Battery current (uA); negative=discharging */
-#define FUEL_GAUGE_CURRENT		FUEL_GAUGE_AVG_CURRENT + 1
+#define FUEL_GAUGE_CURRENT		    FUEL_GAUGE_AVG_CURRENT + 1
```
Comment (aaronemassey) [inline include/zephyr/drivers/fuel_gauge.h:40]: Was this from a `clang-format`? If so, please have that as a separate commit.

**Review by aaronemassey — CHANGES_REQUESTED** [review]

File: drivers/fuel_gauge/sbs_gauge/sbs_gauge.c [hunk 2]
```diff
@@ -95,9 +95,9 @@ static int sbs_gauge_get_prop(const struct device *dev, struct fuel_gauge_get_pr
 		rc = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_VOLTAGE, &val);
 		prop->value.voltage = val * 1000;
 		break;
-	case FUEL_GAUGE_MODE:
```
Comment (aaronemassey) [inline drivers/fuel_gauge/sbs_gauge/sbs_gauge.c:98]: Let's do renaming in a separate PR. I'm happy to care of that too since

**Review by aaronemassey — CHANGES_REQUESTED** [review]

File: include/zephyr/drivers/fuel_gauge.h (hunk 1 above)
Comment (madhurimaparuchuri) [reply include/zephyr/drivers/fuel_gauge.h:40]: Separated in new revision

File: drivers/fuel_gauge/sbs_gauge/sbs_gauge.c (hunk 2 above)
Comment (madhurimaparuchuri) [reply drivers/fuel_gauge/sbs_gauge/sbs_gauge.c:98]: I renamed only at rate properties in a separate commit in this PR as it needs to be done anyways

**Review by aaronemassey — APPROVED** [review]
@teburd Can you also review?

**Review by teburd — APPROVED** [review]
Looks fine to me, first two commits could possibly be squashed into a Add SBS specific attributes for AtRate, etc. Not a blocker.

@rgundi is likely interested in this one
